compiler/rustc_codegen_llvm/src/consts.rs

```diff
@@ -390,7 +390,7 @@ impl<'ll> CodegenCx<'ll, '_> {
             let val_llty = self.val_ty(v);
 
             let g = self.get_static_inner(def_id, val_llty);
-            let llty = self.val_ty(g);
+            let llty = llvm::LLVMGlobalGetValueType(g);
```

Comment:
Can you please also adjust the `self.val_ty(g) != self.type_ptr()` check above? That one is also supposed to read `LLVMGlobalGetValueType(g) != llty` I believe.

Reply:
hm... i found the initializer type is always a `packed struct`(val_ty) at now. which can never be equal to the `llty`.

Reply:
why do we use `cx.const_struct(&llvals, true)` to create a packed const struct  at `const_alloc_to_llvm`. can we create a normal struct for static initializer

Reply:
> hm... i found the initializer type is always a `packed struct`(val_ty) at now. which can never be equal to the `llty`.

But isn't this also the type we pass as llty? It's the type of the initializer `v` above, which should also be a packed struct?

Reply:
the `llty` is the type of old global var, which is set at `predefine_static`. and the `val_llty` is the type of initializer `v`. 
the llty won't be a packed struct